### PR TITLE
Upgrade citeproc to 3.x snapshot without graal

### PR DESCRIPTION
**Pull request by Siedlerchr**
Refs https://github.com/michel-kraemer/citeproc-java/issues/92

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Test on Mac so far sucessful

**Review by koppor — DISMISSED** [review]
- Windows: Starts and CSL works.
- Linux (WSL2): Starts and CSL works.

(Tested using `gradlew run`)

File: build.gradle [hunk 1]
```diff
@@ -567,56 +566,49 @@ jlink {
...
+        requires 'jdk.unsupported';
+        requires 'java.net.http';
+        requires 'com.google.gson';
+        requires 'java.management';
+        requires 'java.naming';
+        requires 'jdk.unsupported.desktop';
+        requires 'java.security.sasl';
+        requires 'java.scripting';
+        requires 'java.datatransfer';
+        requires 'java.compiler';
```
Comment (koppor) [inline build.gradle:587]: Here, everything seems to be changed. Could we have some alphabetical ordering for the non-grouped dependencies?

Comment (Siedlerchr) [reply build.gradle]: now sorted alphabetically

File: src/main/java/org/jabref/gui/maintable/MainTableDataModel.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public MainTableDataModel(BibDatabaseContext context, PreferencesService prefere
         // We need to wrap the list since otherwise sorting in the table does not work
         entriesSorted = new SortedList<>(entriesFiltered);
         groupViewMode = preferencesService.getGroupViewMode();
+
```
Comment (koppor) [inline src/main/java/org/jabref/gui/maintable/MainTableDataModel.java:57]: Sneaky empty line in here? Could you remove it at the PR where you are really working on that?

**Review by koppor — APPROVED** [review]
I will merge to get things going. Reviewing this PR is less important than working on other PRs and issues.

**Pull request merged**